source/opt/mem_pass.cpp

```diff
@@ -183,7 +183,8 @@ void MemPass::AddStores(uint32_t ptr_id, std::queue<ir::Instruction*>* insts) {
   });
 }
 
-void MemPass::DCEInst(ir::Instruction* inst) {
+void MemPass::DCEInst(ir::Instruction* inst,
+                      const function<void(ir::Instruction * )>& call_back) {
```

Comment:
Should we default the callback function to an empty functor?  It's not always used.

Reply:
I think the latest version will always use the call back.  I just push that a bit earlier today.